sup3r/preprocessing/batch_handling.py

```diff
@@ -32,10 +34,10 @@
 
 
 class Batch:
```

Comment:
I think we should be subclassing for most of the changes in here - instead of editing the older classes. You could implement these new classes by subclassing in another file called conditional_moment_batch_handling.py, or something.

Reply:
@grantbuster Do you agree with this^

Reply:
I could go either way, pro: I could see the value in adding the masking logic to the base class, con: adds bloat and overhead compute that may not be used.

Reply:
Maybe the masking could go into the old classes and the cond moment/output stuff could be subclassed.

Reply:
I think that for now I will add the masking only for the conditional moment batching system. We can decide later if we want to add this for all batching :-)